### PR TITLE
fix SVG gradient rendering for iOS WebViews

### LINKED ISSUE #88: Iro.js gradients do not render properly in cordova-plugin-ionic-webview on iOS
This is a continuation from the conversation near the end of #45

**Context**

Safari for iOS/macOS and iOS webviews handle SVG [IRIs](https://www.w3.org/TR/SVG11/linking.html#IRIReference) strangely, and sometimes require references for things like gradients and filters to be absolute urls (for example, instead of `fill="url(#gradient)"`, you need to use `fill="url(https://example.com/some/page#gradient)"`). 

This is detailed more in #18 and there's several stackoverflow threads which explain this better than I can:
https://stackoverflow.com/questions/19742805/angular-and-svg-filters/19753427
https://stackoverflow.com/questions/29736284/url-to-linear-gradient-doesnt-work

To work around this, iro.js uses a helper function to resolve SVG gradient references to absolute URLs if necessary. That function looks like this:

```js
resolveUrl(url) {
  const location = window.location;
  const ua = window.navigator.userAgent;
  const isSafari = /^((?!chrome|android).)*safari/i.test(ua);
  const isIos = /iPhone|iPod|iPad/i.test(ua);
  return ((sSafari || isIos) ? `${location.protocol}//${location.host}${location.pathname}${location.search}${url}` : url;
}
```

TL;DR: it sniffs the user agent string to detect if iro.js is running in Safari or an iOS Webview, if that's the case, then it prepends the provided url with the full location path for the current page to make it an absolute URL. This seems to work fine for most purposes.

**The Problem**

When used iro.js is used in an Ionic app with Cordova and cordova-plugin-ionic-webview, this solution doesn't seem to work on iOS... and I'm not entirely sure why.

I *think* it may be related to cordova-plugin-ionic-webview setting `location.protocol` to `ionic://`, presumably to [work around CORS issues](https://ionicframework.com/docs/faq/cors#ionic-webview-3-x-plugin-on-cordova).

My initial solution was to simply add a check for `location.protocol` being set to `ionic://` and use relative URLs if that was the case, which surprisingly works in `ionic-angular` version 3.9.5, but not version 4.11.1.

My second solution was to use protocol-relative urls (e.g. `//example.com/some/page#gradient`) instead of absolute URLS. Weirdly, this seems to work in ionic-angular version 4.11.1 ... but not 3.9.5?

**References**

@13546777510 has helpfully created some test repositories to reproduce the error:
https://github.com/13546777510/iro-ionic3
https://github.com/13546777510/iro-ionic4

They opened an issue on the main ionic repository, which helped to pinpoint cordova-plugin-ionic-webview as a potential culprit: https://github.com/ionic-team/ionic/issues/19710

They also opened an issue on the cordova-plugin-ionic-webview repository: https://github.com/ionic-team/cordova-plugin-ionic-webview/issues/457

----

I would really appreciate any help with this, since I'm really not very familiar with Ionic, Cordova or Angular myself.

Thanks!

### PR DESCRIPTION
**Pull request by jcesarmobile**
After reading #88 I saw you were fighting the wrong battle. Your problem is not Ionic nor the Ionic WebView, but the `<base href="/" />` tag.

Ionic 4 apps have it, Ionic 3 apps don't. (Probably related to the Angular versions used by those Ionic versions)

So instead of checking for the ionic protocol, which breaks the Ionic 4 apps (and can be configured by the user, BTW), just check for the base tag, as that's what breaks the SVG urls. 

closes #88

### CODE REVIEW

**Comment by jaames** [comment]
Ah, nice catch! 

I had a feeling it was something like that, thanks for the info and PR :)
